### ISSUE TITLE
Error using vl_nnconv

### ISSUE THREAD

**Issue by nh13871101238** [vlfeat/matconvnet#752]
Hello:
I meet some problem when I using  FCN  "trainOpts.gpus = [1] ;",the error like these:

" Error using vl_nnconv
DATA and FILTERS do not have compatible formats.

Error in dagnn.Conv/forward (line 11)
  outputs{1} = vl_nnconv(...
....."
if I using trainOpts.gpus = [],it worked, very strange 
Looking for your help ,thanks!!!

**Comment by foolwood** [comment]
when you get batch 

you should put all input in gpu!!

gpuArray(img)
gpuArray(mask)

I met the similar problem. I think maybe you should check for this.

**Comment by nh13871101238** [comment]
Thanks for you reply !
    I edited my getBatch.m file in line 91,like this ：

   if opts.useGpu
    ims = gpuArray(ims) ;
    labels = gpuArray(labels) ;
   end

but still the same error

**Comment by nh13871101238** [comment]
Any one can help me in fcn ,very important, thanks very much

**Comment by nh13871101238** [comment]
Any one can help me ？？？？？？？？？

**Comment by marksunpeng** [comment]
maybe you can check: data type / gpuarray('cpuarray')
these need to be same format

**Comment by leonlulu** [comment]
Hello nh13871101238, I met the same problem recently. I found that 'opts.useGpu' is set to false in the function 'getBatch'. Changing it to true should solve your problem~

**Comment by mellysa11** [comment]
I have the same problem and tried all the solution above but still got the error. 
@leonlulu in my 'getBatch' function is like this 
function inputs = getBatch(opts, imdb, batch)
    images = imdb.images.data(:,:,:,batch);
    labels = imdb.images.labels(1,batch) ;
    if opts.useGpu > 0
        images = gpuArray(images) ;
    end

    inputs = {'input', images, 'label', labels} ;
end

**Comment by duaisha** [comment]
was anybody able to solve ths problem?
